### PR TITLE
Helm Execute Lego

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
Fixes [EN-5239]

Please include a summary of the change, motivation and context.

This PR brings in support to execute `helm` command. Any helm command that can be executed on command line are supported. 

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

#### Unit Testing

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/99d8906f-7efc-461a-bb52-65ecd9706b4d

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository. 

[EN-5239]: https://unskript.atlassian.net/browse/EN-5239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        print(f"K8S Connector is invalid: {handle}")
+        return str()
+    
+    if not helm_command:
+        print(f"Given helm command is empty, cannot proceed further!")
+        return str()
+    
+    config_file = None
+    try:
+        config_file = handle.temp_config_file
```
Comment (amit-chandak-unskript) [inline Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py:43]: why are we doing this way? Why not define a helm lambda, just like we have for running kubectl command and call that??

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                                    check=True,
+                                    shell=True,
+                                    capture_output=True,
+                                    text=True)
+            retval = result.stdout 
+        except subprocess.CalledProcessError as e:
+            error_message = f"Error running command: {e}\n{e.stderr.decode('utf-8')}" \
+                        if e.stderr else f"Error running command: {e}"
+            print(error_message)
+    else:
```
Comment (amit-chandak-unskript) [inline Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py:63]: Thats not an error case, in incluster case, the config file will be None, in that case you dont need to pass --kubeconfig argument. Please look at how kubectl handles it.

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        print("ERROR: Kubernetes Configuration file is corrupt, cannot execute helm command")
+
```
Comment (amit-chandak-unskript) [inline Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py:65]: Handle the case where in result.stderr is set, in that case, you need to print the error.

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py:43]: As we discussed, will keep the logic of running subprocess in the lego itself.

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py:52]: Thank you for pointing that out, i am handling the case as the second case.

File: Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py (hunk 3 above)
Comment (jayasimha-raghavan-unskript) [reply Kubernetes/legos/k8s_execute_helm_command/k8s_execute_helm_command.py]: Yes, added the check to see if error code (returncode) is not Zero and stderr is set, to print the error.

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**